sway/input/text_input.c

```diff
@@ -255,6 +268,220 @@ static void relay_handle_text_input(struct wl_listener *listener,
 	sway_text_input_create(relay, wlr_text_input);
 }
 
+static void input_popup_update(struct sway_input_popup *popup) {
+	struct sway_text_input *text_input =
+		relay_get_focused_text_input(popup->relay);
+
+	if (text_input == NULL || text_input->input->focused_surface == NULL) {
+		return;
+	}
+
+	if (popup->scene_tree != NULL) {
+		wlr_scene_node_destroy(&popup->scene_tree->node);
+		popup->scene_tree = NULL;
+	}
+	if (popup->desc.relative != NULL) {
+		wlr_scene_node_destroy(popup->desc.relative);
+		popup->desc.relative = NULL;
+	}
+	popup->desc.view = NULL;
+
+	if (!popup->popup_surface->surface->mapped) {
+		return;
+	}
+
+	wlr_scene_node_destroy(&popup->scene_tree->node);
+	wlr_scene_node_destroy(popup->desc.relative);
+	popup->scene_tree = NULL;
+	popup->desc.view = NULL;
```

Comment:
Hm, isn't this crashing now that we set `popup->scene_tree = NULL` above?

I think this whole code block can be removed.